pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -522,12 +522,12 @@ func NewOpenStackClient(cfg *AuthOpts, userAgent string, extraUserAgent ...strin
 		}
 	}
 
+	config := &tls.Config{}
 	if caPool != nil {
-		config := &tls.Config{}
 		config.RootCAs = caPool
-		config.InsecureSkipVerify = cfg.TLSInsecure == "true"
-		provider.HTTPClient.Transport = netutil.SetOldTransportDefaults(&http.Transport{TLSClientConfig: config})
 	}
+	config.InsecureSkipVerify = cfg.TLSInsecure == "true"
+	provider.HTTPClient.Transport = netutil.SetOldTransportDefaults(&http.Transport{TLSClientConfig: config})
```

Comment:
@kayrus If these are moved out of if condition, by default it is applied for all connections , is this what intended?

Reply:
yes, the `SetOldTransportDefaults` also sets transport timeout and no_proxy settings. If the schema is http, then tlsconfig will be just ignored. I'm trying to find the evidence, that a regular http scheme will work fine with these settings, will post it later.